### PR TITLE
Perf set changing prop

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
This PR introduces a new benchmark that focuses on performance of property updates. It also contain a fix where by removing a megamorphic access we can speed up this new benchmark by ~30%. 

Review commit by commit and check individual commit message for more details.

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

File: packages/core/src/render3/instructions/shared.ts [hunk 1]
```diff
@@ -839,17 +839,23 @@ export function generatePropertyAliases(
 
 /**
  * Mapping between attributes names that don't correspond to their element property names.
+ *
+ * Performance note: this function is written as a series of if checks (instead of, say, a property
+ * object lookup) for performance reasons - the series of `if` checks seems to be the fastest way of
+ * mapping property names. Do NOT change without benchmarking.
```
Comment (kara) [inline packages/core/src/render3/instructions/shared.ts]: 👍

**Review by kara — APPROVED** [review]
LGTM

File: packages/core/src/render3/instructions/shared.ts [hunk 2]
```diff
@@ -839,17 +839,23 @@ export function generatePropertyAliases(
...
-const ATTR_TO_PROP: {[name: string]: string} = {
-  'class': 'className',
-  'for': 'htmlFor',
-  'formaction': 'formAction',
-  'innerHtml': 'innerHTML',
-  'readonly': 'readOnly',
-  'tabindex': 'tabIndex',
-};
+function mapPropName(name: string): string {
+  if (name === 'class') return 'className';
```
Comment (mhevery) [inline packages/core/src/render3/instructions/shared.ts]: I wonder if we could make this even faster if we did something like this:
```
const ch = name.charAt(0);
if (ch === 'c' || ch === 'f' || ch === 'i' || ch === 't' || ch === 't') {
  ...
}
```
```

**Comment by matsko** [comment]
Presubmit = http://test/OCL:268268045:BASE:268268068:1568139654616:ce93ff37

Ivy Presubmit = https://test.corp.google.com/ui#id=OCL:268268045:BASE:268268187:1568139689058:797e08d4

**Comment by matsko** [comment]
Presubmit = https://test.corp.google.com/ui#id=OCL:268268045:BASE:268476802:1568219816264:b888bdce

**Pull request closed without merging**